tests/specs/api.e2e.js

```diff
@@ -99,4 +102,205 @@ describe('newrelic api', () => {
       expect(params.message).toEqual('too many free taco coupons')
     })
   })
+
+  describe('finished()', () => {
+    it('records a PageAction when called before RUM message', async () => {
+      const [insResult] = await Promise.all([
+        browser.testHandle.expectIns(),
+        browser.testHandle.expectRum(),
+        browser.url(await browser.testHandle.assetURL('api/finished.html', {
+          init: {
+            page_view_timing: {
+              enabled: false
+            },
+            ins: {
+              harvestTimeSeconds: 2
+            }
+          }
+        }))
+          .then(() => browser.waitForAgentLoad())
+      ])
+
+      const pageActions = insResult.request.body.ins
+      expect(pageActions).toBeDefined()
+      expect(pageActions.length).toEqual(1) // exactly 1 PageAction was submitted
+      expect(pageActions[0].actionName).toEqual('finished') // PageAction has actionName = finished
```

Comment:
I have found expecting on a specific indexed event to be flakey. Could just do a quick search for the event to see if it was included in the request body.

```suggestion
      const finishedAction = Array.from(insResult.request.body.ins || [])
        .find(e => e.actionName === 'finished')
      expect(finishedAction).toBeDefined()
```

Reply:
What kind of flakey have you seen with expecting specific indexed events? Like the number of items in `.ins` might vary? Whoever wrote this test originally noted that they were expecting exactly one PageAction, which sounds like a stable expectation worth testing for.

Reply:
Yea if there is expected to only be one item in `ins`, that makes sense.